util-cgroups/src/measure/v2/settings.rs

```diff
@@ -303,4 +303,452 @@ mod tests {
         let keys = settings.enabled_keys().unwrap();
         assert_eq!(keys, vec!["key1"]);
     }
+
+    #[test]
+    fn test_serialize_bool() {
+        let serializer = EnabledKeysSerializer;
+        let true_serialized_res = serializer.serialize_bool(true);
+        assert!(true_serialized_res.is_ok());
+        let true_serialized = true_serialized_res.unwrap();
+        match true_serialized {
+            SerializedValue::Flag(bool) => assert_eq!(bool, true),
+            SerializedValue::EnabledSet(_) => assert!(false),
+        }
+
+        let serializer = EnabledKeysSerializer;
+        let false_serialized_res = serializer.serialize_bool(false);
+        assert!(false_serialized_res.is_ok());
+        let false_serialized = false_serialized_res.unwrap();
+        match false_serialized {
+            SerializedValue::Flag(bool) => assert_eq!(bool, false),
+            SerializedValue::EnabledSet(_) => assert!(false),
+        }
+    }
+
+    #[test]
+    fn test_serialize_i8() {
```

Comment:
This can be a single unit test `test_unsupported_types` with assertions compacted to:
```
assert!(matches!(serializer.serialize_i8(1), Err(SerializationError::UnsupportedType)));
```
No need to write thousands of lines for that ^^